cookbook/how_to_replace_inline-javascript_with_component.rst

```diff
@@ -0,0 +1,267 @@
+How to Replace Inline-Javascript with a Component
+=================================================
+
+Embedding Functionality in a Page
+---------------------------------
+
+Commonly, inline-JavaScript is used to bind some interactive functionality into a particular markup
+fragment:
+
+.. code-block:: html
+
+    <select id="my-select">
+        <option value="foo">Foo</option>
+        <option value="bar">Bar</option>
+    </select>
+    <script type="text/javascript">
+        require(['jquery', 'jquery.select2'], function ($) {
+            $('#my-select').select2({
+                placeholder: 'Select one ...',
+                allowClear: true
+            });
+        });
+    </script>
+
+The inline-script is often larger than in the example above and may also make use of inlined Twig
+code. It is impossible to reuse those code, to extend it or to test it. Additionally, it's hard to
+maintain it.
+
+Furthermore, the lifecycle of its instances is not defined and it is not specified how the function
+will be disabled when the control is not used anymore. Instead, one has to rely on jQuery to
+properly clean the memory. Most of the time, jQuery does this fine. However, there's no absolute
+guarantee that jQuery always handles this task successfully without the developer's help.
+
+The solution for the problems explained above is a Page Component.
+
+The Page Component
+------------------
+
+A Page Component is a controller that is not used for a whole page of an application, but will be
+used to implement certain parts of its functionality. A Page Component is responsible for the
+following things:
+
+ * creating related views, collections, models and even sub-components
+ * handling environment events
+ * disposing obsolete instances
+
+.. seealso::
+
+    You can find more information about the Page Component in the
+    :doc:`"Frontend Architecture chapter" </book/frontend_architecture>` and in the
+    `Page Component documentation`_.
+
+Creating a Page Component Module
+--------------------------------
+
+A Page Component is a JavaScript object based on the ``BaseComponent``. The inline-JavaScript from
```

Comment:
Think, it is fine. Thanks.
It's hard to make better a sentence without ``class`` word, where ``class`` is the best fit for it.